utils/zerovec/src/zerovec/mod.rs

```diff
@@ -392,6 +392,65 @@ where
         self.as_slice().iter().map(T::from_unaligned)
     }
 
+    /// Mutates each element according to a given function, meant to be
+    /// a more convenient version of [`ZeroVec::iter_mut()`] which serves fewer use cases.
+    ///
+    /// This will convert the ZeroVec into an owned ZeroVec if not already the case.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    /// use zerovec::ule::AsULE;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let mut zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    ///
+    /// zerovec.for_each_mut(|item| *item += 1);
+    ///
+    /// assert_eq!(zerovec.to_vec(), &[212, 282, 422, 462]);
+    /// ```
+    #[inline]
+    pub fn for_each_mut(&mut self, mut f: impl FnMut(&mut T)) {
+        self.make_mut().iter_mut().for_each(|item| {
+            let mut aligned = T::from_unaligned(item);
+            f(&mut aligned);
+            *item = aligned.as_unaligned()
+        });
+    }
+
+    /// Same as [`ZeroVec::for_each_mut()`], but bubbles up errors.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use zerovec::ZeroVec;
+    /// use zerovec::ule::AsULE;
+    ///
+    /// let bytes: &[u8] = &[0xD3, 0x00, 0x19, 0x01, 0xA5, 0x01, 0xCD, 0x01];
+    /// let mut zerovec: ZeroVec<u16> = ZeroVec::parse_byte_slice(bytes).expect("infallible");
+    ///
+    /// zerovec.try_for_each_mut(|item| {
+    ///     *item = item.checked_add(1).ok_or(())?;
+    ///     Ok(())   
+    /// })?;
+    ///
+    /// assert_eq!(zerovec.to_vec(), &[212, 282, 422, 462]);
+    /// # Ok::<(), ()>(())
```

Comment:
what is this line for?  Is this to make the `?` operator work?

Reply:
Yes. It's not shown in the docs, just makes the doctest run